File: frontend/src/microservices/hotel/objects/HotelRoomTypeImageUploadResponse.ts
// 本文件定义 hotel 模块的 `HotelRoomTypeImageUploadResponse`，作为响应数据并提供 JSON 编解码。

export type HotelRoomTypeImageUploadResponse = {
  assetId: string
  publicUrl: string
  originalFileName: string
  mimeType: string
  fileSize: number
}

export const hotelRoomTypeImageUploadResponseFromJson = (json: string): HotelRoomTypeImageUploadResponse =>
  JSON.parse(json) as HotelRoomTypeImageUploadResponse

export const hotelRoomTypeImageUploadResponseToJson = (value: HotelRoomTypeImageUploadResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/HotelRoomTypeImageUploadResponse.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class HotelRoomTypeImageUploadResponse(
    assetId: String,
    publicUrl: String,
    originalFileName: String,
    mimeType: String,
    fileSize: Long
)

object HotelRoomTypeImageUploadResponse:
  given sourceEncoder: Encoder[HotelRoomTypeImageUploadResponse] = deriveEncoder
  given sourceDecoder: Decoder[HotelRoomTypeImageUploadResponse] = deriveDecoder
